symtabAPI/src/Type.C

```diff
@@ -296,42 +296,12 @@ bool Type::isCompatible(Type * /*oType*/)
 /*
  * ENUM
  */
-typeEnum::typeEnum(int ID, std::string name)
-    : Type(name, ID, dataEnum)
-{
-	size_ = sizeof(int);
-}
-
-typeEnum::typeEnum(std::string name)
-   : Type(name, ::getUniqueTypeId(), dataEnum)
-{
-   size_ = sizeof(int);
-}
-
-typeEnum *typeEnum::create(std::string &name, dyn_c_vector< std::pair<std::string, int> *> &constants, Symtab *obj)
-{
-   typeEnum *typ = new typeEnum(name);
-   for(unsigned i=0; i<constants.size();i++)
-   	typ->addConstant(constants[i]->first, constants[i]->second);
-    
-    if(obj)
-       obj->addType(typ);
-    //obj->addType(typ); TODO: declare a static container if obj is NULL and add to it.
-    //Symtab::noObjTypes->push_back(typ); ??
-    return typ;	
+typeEnum::typeEnum(boost::shared_ptr<Type> underlying_type, std::string name, typeId_t ID) :
+		derivedType(name, ID, underlying_type->getSize(), dataEnum) {
+	baseType_ = underlying_type;
```

Comment:
why isn't underlying_type passed to derivedType's constructor, and that constructor initializes baseType_?

Reply:
I forgot to make a note about that. It's because there is no constructor for `derivedType` that can do that (https://github.com/dyninst/dyninst/blob/master/symtabAPI/h/Type.h#L365).